test/functional/p2p_compactblocks.py

```diff
@@ -730,6 +730,10 @@ def test_invalid_tx_in_compactblock(self, test_node):
         block.vtx[0].wit.vtxinwit = []
         block.solve()
 
+        # Make sure node has the transactions to reconstruct the block
+        for i in range(1, len(block.vtx)):
```

Comment:
all these transactions being sent are in the `prefill_list` which means its in the compact block message itself, can you explain what you're solving here?

Reply:
to be honest I'm confused as to what the test is trying to cover as-is, it's introducing 2 consensus errors, the one that is detected first is a coinbase witness issue

Reply:
> all these transactions being sent are in the prefill_list which means its in the compact block message itself, can you explain what you're solving here?

What this test (as-is) is doing is basically:
1. Create a block but dropping the coinbase witness and including the witness commitment.
2. Send the compact block with all transactions prefilled, and verify that we don't get disconnected.
3. Check that the tip does not advance

The problem is, regardless the step 1, the tip will not advance anyway because, even if the block being valid, since the receiver does not have all transactions, it will request the missing ones back and until it doesn't have all of them, the tip will not advance anyway. In this PR, I send the full txs before sending the compact block, so we can make sure the tip will not really advance due to "step 1".

Reply:
fyi the block also deleted the 4th tx, making the final transaction invalid since they're a chain. The last tx or two are simply invalid and won't be in the mempool anyways.

This fails:
```
diff --git a/test/functional/p2p_compactblocks.py b/test/functional/p2p_compactblocks.py
index 274ef9532c..7b4eb8f986 100755
--- a/test/functional/p2p_compactblocks.py
+++ b/test/functional/p2p_compactblocks.py
@@ -731,10 +731,11 @@ class CompactBlocksTest(BitcoinTestFramework):
         block.solve()
 
         # Make sure node has the transactions to reconstruct the block
         for i in range(1, len(block.vtx)):
             test_node.send_and_ping(msg_tx(block.vtx[i]))
+            assert block.vtx[i].sha256 in node.getrawmempool()
 
         # Now send the compact block with all transactions prefilled, and
         # verify that we don't get disconnected.
         comp_block = HeaderAndShortIDs()
         comp_block.initialize_from_block(block, prefill_list=[0, 1, 2, 3, 4], use_witness=True)
```

> Send the compact block with all transactions prefilled, and verify that we don't get disconnected.

is contradicted by 

> since the receiver does not have all transactions

?

My logs suggest that all the transactions are found too, since they're passing merkle checks:

"2024-12-05T17:07:10.895018Z [msghand] [validation.cpp:4594] [ProcessNewBlock] [error] ProcessNewBl
ock: AcceptBlock FAILED (bad-witness-nonce-size, CheckWitnessMalleation : invalid witness reserved
 value size)"

with the coinbase malleation removed even more clear:

`2024-12-05T17:11:05.392987Z [msghand] [validationinterface.cpp:249] [BlockChecked] [validation] BlockChecked: block hash=34f96e2a2d2fe4ed23eb09f39a770a1946a22aac30ce46371c57473f26b52cfb state=bad-txns-inputs-missingorspent, CheckTxInputs: inputs missing/spent`

Reply:
If you mean you want something more easily verifiable in case the block is *actually correct* somehow, then just change the prefill arg to make it the whole block?

```
prefill_list=list(range(len(block.vtx)))
```

Reply:
I think it's just an a off-by-one error in master, which is best solved with @instagibbs suggestion.
The intent was to send all 5 txns prefilled ("Now send the compact block with all transactions prefilled"), but it would actually send only 4, because the original author forgot when writing `prefill_list=[0, 1, 2, 3, 4]` that there is also the coinbase tx.

Reply:
> If you mean you want something more easily verifiable in case the block is actually correct somehow, then just change the prefill arg to make it the whole block?

Yes, sorry for the confusion. "Send the compact block with all transactions prefilled" is basically the necessary transactions for the block, I meant ALL in fact. As-is, if we don't delete `vtx[3]` then it would request it back and the tip wouldn't advance. I'll take your suggestion.

Reply:
Done.